tools/generator/src/Extensions/XCScheme+Extensions.swift

```diff
@@ -0,0 +1,227 @@
+                customLLDBInitFile: XCSchemeConstants.customLLDBInitFile
+            )
+        }
+
+        let launchAction: XCScheme.LaunchAction?
+        if let launchActionInfo = schemeInfo.launchActionInfo {
+            launchAction = try .init(buildMode: buildMode, launchActionInfo: launchActionInfo)
+        } else {
+            launchAction = .init(
+                runnable: nil,
+                buildConfiguration: XCSchemeConstants.defaultBuildConfigurationName,
+                customLLDBInitFile: XCSchemeConstants.customLLDBInitFile
+            )
+        }
+
+        let profileAction: XCScheme.ProfileAction?
+        if let profileActionInfo = schemeInfo.profileActionInfo {
+            profileAction = .init(profileActionInfo: profileActionInfo)
+        } else {
+            profileAction = .init(
+                buildableProductRunnable: nil,
+                buildConfiguration: XCSchemeConstants.defaultBuildConfigurationName
+            )
+        }
+
+        self.init(
+            name: schemeInfo.name,
+            lastUpgradeVersion: XCSchemeConstants.defaultLastUpgradeVersion,
+            version: XCSchemeConstants.lldbInitVersion,
+            buildAction: buildAction,
+            testAction: testAction,
+            launchAction: launchAction,
+            profileAction: profileAction,
+            analyzeAction: .init(analyzeActionInfo: schemeInfo.analyzeActionInfo),
+            archiveAction: .init(archiveActionInfo: schemeInfo.archiveActionInfo),
+            wasCreatedForAppExtension: schemeInfo.wasCreatedForAppExtension ? true : nil
+        )
+    }
+}
+
+// MARK: XCScheme.BuildAction
+
+extension XCScheme.BuildAction {
+    convenience init(
+        buildMode: BuildMode,
+        buildActionInfo: XCSchemeInfo.BuildActionInfo
+    ) throws {
+        self.init(
+            buildActionEntries: buildActionInfo.targetInfos.buildActionEntries,
+            preActions: buildMode.usesBazelModeBuildScripts ?
+                try buildActionInfo.targetInfos.bazelBuildPreActions : [],
+            parallelizeBuild: true,
+            buildImplicitDependencies: true
+        )
+    }
+}
+
+// MARK: XCScheme.BuildAction.Entry
+
+extension XCScheme.BuildAction.Entry {
+    convenience init(withDefaults buildableReference: XCScheme.BuildableReference) {
+        self.init(
+            buildableReference: buildableReference,
+            buildFor: [
+                .running,
+                .testing,
+                .profiling,
+                .archiving,
+                .analyzing,
+            ]
+        )
+    }
+}
+
+// MARK: XCScheme.ExecutionAction
+
+extension XCScheme.ExecutionAction {
+    /// Initialize the output file for build with Bazel
+    static let initBazelBuildOutputGroupsFile = XCScheme.ExecutionAction(
+        scriptText: #"""
+mkdir -p "${BAZEL_BUILD_OUTPUT_GROUPS_FILE%/*}"
+if [[ -s "$BAZEL_BUILD_OUTPUT_GROUPS_FILE" ]]; then
+    rm "$BAZEL_BUILD_OUTPUT_GROUPS_FILE"
+fi
+"""#,
```

Comment:
This needs a trailing newline.
```suggestion
        scriptText: #"""
mkdir -p "${BAZEL_BUILD_OUTPUT_GROUPS_FILE%/*}"
if [[ -s "$BAZEL_BUILD_OUTPUT_GROUPS_FILE" ]]; then
    rm "$BAZEL_BUILD_OUTPUT_GROUPS_FILE"
fi

"""#,
```

Reply:
Done